lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -374,7 +376,13 @@ public function query()
             $logger->startQuery($args[0]);
         }
 
-        $statement = $this->_conn->query(...$args);
+        try {
+            $statement = $this->_conn->query(...$args);
+        } catch (\Throwable $ex) {
```

Comment:
We probably need to catch a more specific type of exception here. Whatever issues unrelated to the connection failure should not be converted to the driver exception (e.g. all types of `Error`).